client/layout/masterbar/plan-upsell.jsx

```diff
@@ -0,0 +1,96 @@
+import PropTypes from 'prop-types';
+import { createRef, Component } from 'react';
+import { connect } from 'react-redux';
+import TranslatableString from 'calypso/components/translatable/proptype';
+import { ProvideExperimentData } from 'calypso/lib/explat';
+import { recordTracksEvent } from 'calypso/state/analytics/actions';
+import getPrimarySiteId from 'calypso/state/selectors/get-primary-site-id';
+import isSiteWPForTeams from 'calypso/state/selectors/is-site-wpforteams';
+import { getCurrentPlan } from 'calypso/state/sites/plans/selectors';
+import { getSiteSlug, isJetpackSite } from 'calypso/state/sites/selectors';
+import { getSelectedSiteId } from 'calypso/state/ui/selectors';
+import MasterbarItem from './item';
+
+class MasterbarItemPlanUpsell extends Component {
+	static propTypes = {
+		className: PropTypes.string,
+		tooltip: TranslatableString,
+	};
+
+	planUpsellButtonRef = createRef();
+
+	clickPlanUpsell = () => {
+		this.props.recordTracksEvent( 'calypso_masterbar_click', {
+			clicked: 'plan_upsell_button',
+		} );
+	};
+
+	render() {
+		const { siteSlug, isP2, className, isJetpack, tooltip, planSlug } = this.props;
+
+		const plansUpsells = [
+			'free_plan',
+			'personal-bundle-monthly',
+			'value_bundle_monthly',
+			'business-bundle-monthly',
+			'ecommerce-bundle-monthly',
```

Comment:
If it's an eCommerce monthly site, then can we take the Upgrade click to the annual plans tab of `/plans` page? Otherwise the monthly plans tab will show them to be on the highest tier.